src/main/scala/it/pagopa/interop/agreementprocess/api/impl/AgreementApiServiceImpl.scala

```diff
@@ -204,8 +204,7 @@ final case class AgreementApiServiceImpl(
       requesterOrgId <- getOrganizationIdFutureUUID(contexts)
       agreementUUID  <- agreementId.toFutureUUID
       agreement      <- agreementManagementService.getAgreementById(agreementUUID)
-      _              <- assertRequesterIsProducer(requesterOrgId, agreement)
-      // Correct way to check if agreement is in draft state?
+      _              <- assertRequesterIsConsumer(requesterOrgId, agreement)
```

Comment:
Are you sure that a producer should never call this API? Maybe we should consider using the condition `assertRequesterIsConsumer(requesterOrgId, agreement) || assertRequesterIsProducer(requesterOrgId, agreement)`, or maybe we should consider the chance to let the producer and the consumer update an agreement differently.

Reply:
Maybe we can use assertRequesterIsConsumerOrProducer() that is already present. From a talk with "frontend" only consumers needed to do this operation. @galales @beetlecrunch

Reply:
I can confirm that this api should not be called by the producer 👍